src/DataTypes/DataTypeEnum.h

```diff
@@ -80,13 +81,25 @@ class DataTypeEnum final : public IDataTypeEnum
         return findByValue(value)->second;
     }
 
-    FieldType getValue(StringRef field_name) const
+    FieldType getValue(StringRef field_name, bool try_treat_as_id=false) const
     {
-        const auto it = name_to_value_map.find(field_name);
-        if (!it)
+        const auto value_it = name_to_value_map.find(field_name);
+        if (!value_it)
+        {
+            /// It is used in CSV and TSV input formats. If we fail to find given string in
+            /// enum names and this string is number, we will try to treat it as enum id.
+            if (try_treat_as_id && isStringNumber(field_name))
+            {
+                FieldType x;
+                ReadBufferFromMemory tmp_buf(field_name.data, field_name.size);
+                readText(x, tmp_buf);
```

Comment:
Better implementation possible. Here we process field_name twice (first to check if it contains only digits, second time to parse the number).

Instead we can call readText and check that whole string was read (tmp_buf.count()).